src/block.rs

```diff
@@ -1,33 +1,314 @@
 
-#[derive(Debug)]
+use std::mem::transmute;
+use serde::{Serialize, Deserialize};
+
+use data_encoding::HEXLOWER;
+
+use crate::burnfee::BurnFee;
+use crate::crypto::{PublicKey, hash};
+use crate::transaction::Transaction;
+
+use crate::helper::create_timestamp;
+/// Our Block object
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
 pub struct Block {
-    
+    pub body:     BlockBody,
+    pub is_valid: u8,
+    mintid:   u32,
+    maxtid:   u32,
+    bsh:      [u8; 32],
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct BlockBody {
+    pub id:          u32,
+    pub ts:          u64,
+    pub prevbsh:     [u8; 32],
+    pub creator:     PublicKey,
+    pub txs:         Vec<Transaction>,
+    pub bf:	         BurnFee,
+    merkle:          [u8; 32],
+    difficulty:      f32,
+    paysplit:        f32,
+    vote:            i8,
+    treasury:        u64,
+    coinbase:        u64,
+    reclaimed:       u64
+}
+
+impl BlockBody {
+    pub fn new(block_creator: PublicKey, prevbsh: [u8;32]) -> BlockBody {
+        return BlockBody {
+    	    id:          0,
+    	    ts:          create_timestamp(),
+    	    prevbsh:     prevbsh,
+    	    merkle:      [0; 32],
+    	    creator:     block_creator,
+    	    txs:         vec![],
+	        bf:          BurnFee::new(0.0, 0),
+    	    difficulty:  2.1,
+    	    paysplit:    0.5,
+    	    vote:        0,
+    	    treasury:    286_810_000_000_000_000,
+    	    coinbase:    0,
+    	    reclaimed:   0
+        };
+    }
 }
 
 impl Block {
-    fn new() -> Block {
-        Block {}
-    }
-    fn deserialize(stream: AsyncRead) -> Block {
-        Block {}
-    }
-    fn serialize() -> AsyncRead {
-        let retArr: [u8] = [];
-        
-        retArr
-    }
-    fn validate()
-    fn addTransaction(Transaction)
-    fn txCount() -> int
-    fn getTransactionByHash(hash) -> Transaction
-    fn getTransactionById(int id) -> Transaction
-    fn getTransactions() -> [Transaction]
-    fn getHash() -> String
-    fn hasTransaction(String hash) -> bool
-    fn attemptToFindDifficultHash() -> Option<String>
+    pub fn new(creator: PublicKey, prevbsh: [u8;32]) -> Block {
+        return Block {
+            body:      BlockBody::new(creator, prevbsh),
+            is_valid:  1,
+            mintid:    0,
+            maxtid:    0,
+            bsh:       [0; 32],
+        };
+    }
+
+    pub fn header(&self) -> BlockHeader {
+        return BlockHeader::new(
+            self.get_bsh(),
+            self.body.prevbsh,
+            self.body.id,
+            self.body.ts,
+            self.body.bf.clone(),
+            self.mintid,
+            self.maxtid,
+            self.body.difficulty,
+            self.body.paysplit,
+            self.body.vote,
+            self.body.treasury,
+            self.body.coinbase,
+            self.body.reclaimed,
+        );
+    }
+
+    // pub fn get_transactions(&mut self, tx: Transaction) -> Vec<Transaction> {
```

Comment:
Comment why this is commented out? Otherwise this will stay forever and people forget about it